tests/python_package_test/test_sklearn.py

```diff
@@ -80,1058 +80,1072 @@ def multi_logloss(y_true, y_pred):
-        fit_and_check(['valid_0'], ['l1', 'l2'], iter_min_valid1, False)
-        fit_and_check(['valid_0'], ['l1', 'l2'], iter_valid1_l1, True)
-
-        params_fit['eval_metric'] = ["l2", "l1"]
-        fit_and_check(['valid_0'], ['l1', 'l2'], iter_min_valid1, False)
-        fit_and_check(['valid_0'], ['l1', 'l2'], iter_valid1_l2, True)
-
-        params_fit['eval_metric'] = ["l2", "regression", "mse"]  # test aliases
-        fit_and_check(['valid_0'], ['l2'], iter_valid1_l2, False)
-        fit_and_check(['valid_0'], ['l2'], iter_valid1_l2, True)
-
-        # two eval_set
-        params_fit['eval_set'] = [(X_test1, y_test1), (X_test2, y_test2)]
-        params_fit['eval_metric'] = ["l1", "l2"]
-        fit_and_check(['valid_0', 'valid_1'], ['l1', 'l2'], iter_min_l1, True)
-        params_fit['eval_metric'] = ["l2", "l1"]
-        fit_and_check(['valid_0', 'valid_1'], ['l1', 'l2'], iter_min_l2, True)
-
-        params_fit['eval_set'] = [(X_test2, y_test2), (X_test1, y_test1)]
-        params_fit['eval_metric'] = ["l1", "l2"]
-        fit_and_check(['valid_0', 'valid_1'], ['l1', 'l2'], iter_min, False)
-        fit_and_check(['valid_0', 'valid_1'], ['l1', 'l2'], iter_min_l1, True)
-        params_fit['eval_metric'] = ["l2", "l1"]
-        fit_and_check(['valid_0', 'valid_1'], ['l1', 'l2'], iter_min, False)
-        fit_and_check(['valid_0', 'valid_1'], ['l1', 'l2'], iter_min_l2, True)
-
-    def test_class_weight(self):
-        X, y = load_digits(n_class=10, return_X_y=True)
-        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
-        y_train_str = y_train.astype('str')
-        y_test_str = y_test.astype('str')
-        gbm = lgb.LGBMClassifier(n_estimators=10, class_weight='balanced', silent=True)
-        gbm.fit(X_train, y_train,
-                eval_set=[(X_train, y_train), (X_test, y_test), (X_test, y_test),
-                          (X_test, y_test), (X_test, y_test)],
-                eval_class_weight=['balanced', None, 'balanced', {1: 10, 4: 20}, {5: 30, 2: 40}],
+        assert len(gbm.evals_result_) == len(eval_set_names)
+        for eval_set_name in eval_set_names:
+            assert eval_set_name in gbm.evals_result_
+            assert len(gbm.evals_result_[eval_set_name]) == len(metric_names)
+            for metric_name in metric_names:
+                assert metric_name in gbm.evals_result_[eval_set_name]
+
+                actual = len(gbm.evals_result_[eval_set_name][metric_name])
+                expected = assumed_iteration + (params_fit['early_stopping_rounds']
+                                                if eval_set_name != 'training'
+                                                and assumed_iteration != gbm.n_estimators else 0)
+                assert expected == actual
+                assert (assumed_iteration
+                        if eval_set_name != 'training' else gbm.n_estimators == gbm.best_iteration_)
```

Comment:
Please make this assert more readable by splitting into two asserts.
```suggestion
                if eval_set_name != 'training':
                    assert assumed_iteration == gbm.best_iteration_
                else:
                    assert gbm.n_estimators == gbm.best_iteration_
```